File: LL/src/Core/Application/UseCases/Leaderboards/Dtos/LeaderboardDto.cs
using Application.Common.Mappings;
using AutoMapper;
using Domain.Models.Leaderboards;

namespace Application.UseCases.Leaderboards.Dtos;
public class LeaderboardDto : IMapFrom<Leaderboard>
{
    public List<LeaderboardEntryDto> Combat { get; set; } = [];
    public List<LeaderboardEntryDto> Wealth { get; set; } = [];
    public List<LeaderboardEntry> TotalLevel { get; set; } = [];
    public Dictionary<string, List<LeaderboardEntryDto>> Professions { get; set; } = [];
    public void Mapping(Profile profile)
    {
        profile.CreateMap<Leaderboard, LeaderboardDto>();
    }
}

File: LL/src/Presentation/ll/src/app/shared/models/Dtos/leaderboard/leaderboardDto.ts
import { LeaderboardEntryDto } from './leaderboardEntryDto';

export interface LeaderboardDto {
  totalLevel: LeaderboardEntryDto[];
  combat: LeaderboardEntryDto[];
  wealth: LeaderboardEntryDto[];
  professions: Record<string, LeaderboardEntryDto[]>;
}
